relay-server/src/utils/rate_limits.rs

```diff
@@ -520,6 +520,39 @@ impl Enforcement {
     }
 }
 
+/// Which limits to apply with the [`EnvelopeLimiter`].
+#[derive(Debug, Copy, Clone)]
+pub enum ApplyLimits {
```

Comment:
nit on naming: We have other places where `apply` means drop data and emit outcomes, so should we rename this to `CheckLimits` and `Check::apply` to `Check::check`?